rflx/pyrflx/typevalue.py

```diff
@@ -204,34 +218,36 @@ def assign(self, value: str, check: bool = True) -> None:
             .simplified()
             == TRUE
         )
-        self._value = value
+        self._value = value, self._type.literals[value]
 
     def parse(self, value: Union[Bitstring, bytes]) -> None:
         if isinstance(value, bytes):
             value = Bitstring.from_bytes(value)
-        value_as_int: int = int(value)
-        if not Number(value_as_int) in self.literals.values():
-            raise KeyError(f"Number {value_as_int} is not a valid enum value")
-
-        for k, v in self.literals.items():
-            if v == Number(value_as_int):
-                self._value = str(k)
+        value_as_number = Number(int(value))
+        if value_as_number not in self.literals.values() and not self._type.always_valid:
+            raise KeyError(f"Number {value_as_number.value} is not a valid enum value")
+        if value_as_number not in self.literals.values() and self._type.always_valid:
+            self._value = "UNKNOWN", value_as_number
+        else:
```

Comment:
Could this be rewritten as
```Python
if value_as_number not in self.literals.values():
    if self._type.always_valid:
        self._value = "UNKNOWN", value_as_number
    else:
        raise KeyError(f"Number {value_as_number.value} is not a valid enum value")
else:
```
?
I think this would make the branches more clear.

Reply:
Fixed by b3ec99d